core/src/cluster_info_repair_listener.rs

```diff
@@ -0,0 +1,683 @@
+            .expect("Couldn't get db iterator");
+        let mut total_data_blobs_sent = 0;
+        let mut total_coding_blobs_sent = 0;
+
+        for (slot, slot_meta) in slot_iter {
+            if slot > my_root {
+                break;
+            }
+            if !repairee_epoch_slots.slots.contains(&slot) {
+                // Calculate the blob indexes this node is responsible for repairing. Note that because we
+                // are only repairing slots that are before our root, the slot.received should be equal to
+                // the actual total number of blobs in the slot. Optimistically this means that most repairmen should
+                // observe the same "total" number of blobs for a particular slot, and thus the calculation in
+                // calculate_my_repairman_index_for_slot() will divide responsibility evenly across the cluster
+                let num_blobs_in_slot = slot_meta.received as usize;
+                if let Some((start_index, range_size)) = Self::calculate_my_repairman_index_for_slot(
+                    my_id,
+                    &eligible_repairmen,
+                    num_blobs_in_slot,
+                    REPAIR_REDUNDANCY,
+                ) {
+                    // Repairee is missing this slot, send them the blobs for this slot
+                    for i in 0..range_size {
+                        let mut should_sleep = false;
+                        // Loop over the blob indexes and query the database for these blob that
+                        // this node is reponsible for repairing. This should be faster than using
+                        // a database iterator over the slots because by the time this node is
+                        // sending the blobs in this slot for repair, we expect these slots
+                        // to be full.
+                        let blob_index = (start_index + i) % num_blobs_in_slot;
+                        if let Some(blob_data) = blocktree
+                            .get_data_blob_bytes(slot, blob_index as u64)
+                            .expect("Failed to read data blob from blocktree")
+                        {
+                            socket.send_to(&blob_data[..], repairee_tvu)?;
+                            total_data_blobs_sent += 1;
+                            should_sleep = true;
+                        }
+
+                        if let Some(coding_bytes) = blocktree
+                            .get_coding_blob_bytes(slot, blob_index as u64)
+                            .expect("Failed to read coding blob from blocktree")
+                        {
+                            socket.send_to(&coding_bytes[..], repairee_tvu)?;
+                            total_coding_blobs_sent += 1;
+                            should_sleep = true;
+                        }
+
+                        if should_sleep {
+                            sleep(Duration::from_millis(BLOB_SEND_SLEEP_MILLIS as u64));
```

Comment:
this sleep is per-slot?  if you only do one slot, you can remove this code.

Reply:
This sleep would be per-blob, and the repairman is likely to do many slots in this function (iterates through `slot_iter`)

Reply:
hm.  I think drop this for now